R/functions.R

```diff
@@ -0,0 +1,345 @@
+  # read the planning data
+  read_sheet(
+    ss,
+    sheet = "Planning_v2",
+    col_types = "cccclllilccccccdddddddddddccdddddddddddcc",
+    .name_repair = "minimal"
+  ) |>
+    # clean planning data and turn it into long format
+    clean_names() |>
+    select(
+      -uitvoerders,
+      -reviewers,
+      -starts_with("totaal_dagen"),
+      -ends_with("_buiten_kern"),
+      -starts_with("tijdsinvestering")
+    ) |>
+    rename_with(
+      .cols = karen:datamanager,
+      .fn = \(x) {
+        str_c(x, "_1")
+      }
+    ) |>
+    mutate(
+      continuous = deadline == "Doorlopend",
+      deadline = ifelse(
+        deadline == "Doorlopend",
+        paste0(max_y(), "-12"),
+        deadline
+      ),
+      # intended dates
+      start = ym(start),
+      deadline = ym(deadline) + months(1) - days(1),
+      # actual months (as dates) of starting and finishing (only for started tasks)
+      gestart = ym(gestart),
+      afgerond = ym(afgerond) + months(1) - days(1),
+      # derived variables:
+      # ####################
+      # 2 logicals
+      started = !is.na(gestart),
+      finished = !is.na(afgerond),
+      # first & last day of current month
+      first_day_currentmonth = floor_date(today(), unit = "month"),
+      last_day_currentmonth = ceiling_date(today(), unit = "month") - days(1),
+      # which start date to take in calculating currently applicable interval?
+      begin = case_when(
+        started ~ gestart,
+        !started & finished ~ start,
+        .default = pmax(start, first_day_currentmonth, na.rm = TRUE)
+      ),
+      # which end date to take in calculating currently applicable interval?
+      end = ifelse(finished, afgerond, deadline) |>
+        as.Date(),
+      # is the task overdue?
+      overdue = !finished & today() > end,
+      # number of months of the interval
+      nr_months = case_when(
+        started & overdue ~ as.period(last_day_currentmonth - begin),
+        !started & overdue ~ period(1, "month"),
+        .default = as.period(end - begin)
+      ) |>
+        as.numeric("months") |>
+        round() |>
+        as.integer()
+    ) |>
+    uncount(nr_months, .remove = FALSE) |>
+    mutate(
+      date = begin + months(row_number() - 1),
+      y_month = str_c(year(date), "_", str_pad(month(date), 2, pad = "0")) |>
+        factor(),
+      .by = taakomschrijving
+    ) |>
+    pivot_longer(
+      cols = karen_1:datamanager_2,
+      names_to = c(".value", "task_type"),
+      names_pattern = "(.+)_([12])$"
+    ) |>
+    pivot_longer(
+      karen:datamanager,
+      names_to = "person",
+      values_to = "nr_days",
+      values_drop_na = TRUE
+    ) |>
+    mutate(
+      task_type = ifelse(task_type == "1", "uitvoering", "review") |> fct(),
+      nr_days = ifelse(
+        continuous,
+        nr_days,
+        nr_days / nr_months
+      ),
+      across(where(is.character), fct)
+    ) |>
+    select(-c(first_day_currentmonth:end, nr_months))
+}
```

Comment:
This is my first encounter of a "true" dplyr pipe-line. Quite intimidating... I can read everything, but debugging requires some focus. 
For more public projects, I guess it would be better to split the pipe and document the intermediate steps?
Yet then R fills the stash with those intermediate tibbles, and a gc might be slow to catch up.
Let me know what would be "good practice".

Reply:
"good practice": I'm not too strict here. Some say, like the R4DS book, that a single pipeline should not be longer than 10-15 lines, for readability. Others stress that comments are helpful. I rather incline to the latter advice, i.e. adding comments where things are less self-explanatory (but sometimes forget / underestimate that), only splitting statements based on objects that I need outside the pipeline (indeed in order to limit the number of persistent objects). My personal 'maximum' of lines would rather depend on the level of complexity to grasp what happens during the pipeline.

Anyway I'd have no problem at all with code that has shorter pipelines and is more readable, so feel free on that aspect if you write code.